src/view_element.cpp

```diff
@@ -1488,16 +1488,22 @@ void View::DrawNote(DeviceContext *dc, LayerElement *element, Layer *layer, Staf
             }
             drawingDur = DUR_4;
         }
-        drawingDur = ((note->GetColored() == BOOLEAN_true) && drawingDur > DUR_1) ? (drawingDur + 1) : drawingDur;
-
         if (drawingDur < DUR_BR) {
             this->DrawMaximaToBrevis(dc, noteY, element, layer, staff);
         }
         else {
             // Whole notes
             char32_t fontNo;
             if (note->GetColored() == BOOLEAN_true) {
-                fontNo = (drawingDur == DUR_1) ? SMUFL_E0FA_noteheadWholeFilled : SMUFL_E0A3_noteheadHalf;
+                if (DUR_1 == drawingDur) {
+                    fontNo = SMUFL_E0FA_noteheadWholeFilled;
+                }
+                else if (DUR_2 == drawingDur) {
+                    fontNo = SMUFL_E0A4_noteheadBlack;
```

Comment:
Shouldn't that then be `noteheadHalfFilled`?

Reply:
uhmm, why the half?

Reply:
Because it's a filled half note. In _Leipzig_ the glyph is basically the same as the quarter head, but there are other fonts where they differ. I've added the missing glyph in https://github.com/rism-digital/leipzig/pull/32